### PR TITLE
fixes for zk and property filtering issues

### PR DESCRIPTION
**Pull request by tpetr**
/cc @wsorenson @jhaber

### CODE REVIEW

File: SingularityService/src/main/java/com/hubspot/singularity/guice/GuicePropertyFilteringMessageBodyWriter.java [hunk 1]
```diff
@@ -65,7 +68,7 @@ public void writeTo(final Object o, final Class<?> type, final Type genericType,
 
     final PropertyFiltering annotation = findPropertyFiltering(annotations);
 
-    final PropertyFilter propertyFilter = new PropertyFilter(uriInfo.getQueryParameters().get(annotation.using()));
+    final PropertyFilter propertyFilter = new PropertyFilter(Optional.fromNullable(uriInfo.getQueryParameters().get(annotation.using())).or(Collections.<String>emptyList()));
```
Comment (tpetr) [inline SingularityService/src/main/java/com/hubspot/singularity/guice/GuicePropertyFilteringMessageBodyWriter.java]:
```
WARN  [2015-07-13 15:02:53,191] org.eclipse.jetty.servlet.ServletHandler: /singularity/api/requests/
! java.lang.NullPointerException: null
! at com.hubspot.jackson.jaxrs.PropertyFilter.<init>(PropertyFilter.java:18) ~[SingularityService.jar:0.4.3-SNAPSHOT]
! at com.hubspot.singularity.guice.GuicePropertyFilteringMessageBodyWriter.writeTo(GuicePropertyFilteringMessageBodyWriter.java:68) ~[SingularityService.jar:0.4.3-SNAPSHOT]
! at com.sun.jersey.spi.container.ContainerResponse.write(ContainerResponse.java:302) ~[SingularityService.jar:0.4.3-SNAPSHOT]
! at com.sun.jersey.server.impl.application.WebApplicationImpl._handleRequest(WebApplicationImpl.java:1510) ~[SingularityService.jar:0.4.3-SNAPSHOT]
! at com.sun.jersey.server.impl.application.WebApplicationImpl.handleRequest(WebApplicationImpl.java:1419) ~[SingularityService.jar:0.4.3-SNAPSHOT]
! at com.sun.jersey.server.impl.application.WebApplicationImpl.handleRequest(WebApplicationImpl.java:1409) ~[SingularityService.jar:0.4.3-SNAPSHOT]
! at com.sun.jersey.spi.container.servlet.WebComponent.service(WebComponent.java:409) ~[SingularityService.jar:0.4.3-SNAPSHOT]
! at com.sun.jersey.spi.container.servlet.ServletContainer.service(ServletContainer.java:558) ~[SingularityService.jar:0.4.3-SNAPSHOT]
! at com.sun.jersey.spi.container.servlet.ServletContainer.service(ServletContainer.java:733) ~[SingularityService.jar:0.4.3-SNAPSHOT]
! at javax.servlet.http.HttpServlet.service(HttpServlet.java:790) ~[SingularityService.jar:0.4.3-SNAPSHOT]
```

File: SingularityService/src/main/java/com/hubspot/singularity/data/StateManager.java [hunk 2]
```diff
@@ -39,7 +39,7 @@
   private static final Logger LOG = LoggerFactory.getLogger(StateManager.class);
 
   private static final String ROOT_PATH = "/hosts";
-  private static final String STATE_PATH = "STATE";
+  private static final String STATE_PATH = "/STATE";
```
Comment (tpetr) [inline SingularityService/src/main/java/com/hubspot/singularity/data/StateManager.java]:
```
ERROR [2015-07-13 15:00:59,069] io.dropwizard.jersey.errors.LoggingExceptionMapper: Error handling a request: d5f61c194f1a8e35
! java.lang.IllegalArgumentException: Path must start with / character
! at org.apache.curator.utils.PathUtils.validatePath(PathUtils.java:54) ~[SingularityService.jar:0.4.3-SNAPSHOT]
! at org.apache.curator.utils.PathUtils.validatePath(PathUtils.java:37) ~[SingularityService.jar:0.4.3-SNAPSHOT]
! at org.apache.curator.utils.ZKPaths.fixForNamespace(ZKPaths.java:63) ~[SingularityService.jar:0.4.3-SNAPSHOT]
! at org.apache.curator.framework.imps.NamespaceImpl.fixForNamespace(NamespaceImpl.java:82) ~[SingularityService.jar:0.4.3-SNAPSHOT]
! at org.apache.curator.framework.imps.CuratorFrameworkImpl.fixForNamespace(CuratorFrameworkImpl.java:574) ~[SingularityService.jar:0.4.3-SNAPSHOT]
! at org.apache.curator.framework.imps.GetDataBuilderImpl.forPath(GetDataBuilderImpl.java:270) ~[SingularityService.jar:0.4.3-SNAPSHOT]
! at org.apache.curator.framework.imps.GetDataBuilderImpl.forPath(GetDataBuilderImpl.java:41) ~[SingularityService.jar:0.4.3-SNAPSHOT]
! at com.hubspot.singularity.data.CuratorManager.getData(CuratorManager.java:160) ~[SingularityService.jar:0.4.3-SNAPSHOT]
! at com.hubspot.singularity.data.CuratorManager.getData(CuratorManager.java:176) ~[SingularityService.jar:0.4.3-SNAPSHOT]
! at com.hubspot.singularity.data.StateManager.getState(StateManager.java:122) ~[SingularityService.jar:0.4.3-SNAPSHOT]
! at com.hubspot.singularity.resources.StateResource.getState(StateResource.java:34) ~[SingularityService.jar:0.4.3-SNAPSHOT]
```

File: SingularityService/src/main/docker/start-SingularityService.sh [hunk 3]
```diff
@@ -17,6 +17,7 @@ args+=( -Xmx${SINGULARITY_MAX_HEAP:-512m} )
 args+=( -Djava.net.preferIPv4Stack=true )
 args+=( -Ddw.mesos.master="${SINGULARITY_MESOS_MASTER:=zk://localhost:2181/mesos}" )
 args+=( -Ddw.zookeeper.quorum="${SINGULARITY_ZK:=localhost:2181}" )
+args+=( -Ddw.zookeeper.zkNamespace="${SINGULARITY_ZK_NAMESPACE:=singularity}" )
```
Comment (tpetr) [inline SingularityService/src/main/docker/start-SingularityService.sh]: this wasn't necessarily causing an error, but i figure it'd be good to set / expose in the container

Comment (ssalinas) [inline SingularityService/src/main/docker/start-SingularityService.sh]: :+1:

**Comment by tpetr** [comment]
I believe these issues came in through https://github.com/HubSpot/Singularity/pull/611, which bumped `org.apache.curator:curator-framework` from 2.4.2 to 2.7.1, and `com.hubspot.jackson:jackson-jaxrs-propertyfiltering` from 0.5.0 to 0.7.0.

**Comment by jhaber** [comment]
Ah I see, it reuses the `PropertyFilter` class from `jackson-jaxrs-propertyfiltering` but uses its own `MessageBodyWriter` for some reason, this will probably continue to cause problems

**Pull request merged**